javaslang/src/main/java/javaslang/control/Option.java

```diff
@@ -54,14 +60,14 @@
      */
     static <T> Option<Seq<T>> sequence(Iterable<? extends Option<? extends T>> values) {
         Objects.requireNonNull(values, "values is null");
-        List<T> list = List.empty();
+        Vector<T> vector = Vector.empty();
         for (Option<? extends T> value : values) {
             if (value.isEmpty()) {
                 return Option.none();
             }
-            list = list.prepend(value.get());
+            vector = vector.append(value.get());
         }
-        return Option.some(list.reverse());
+        return Option.some(vector);
     }
```

Comment:
I've double-checked all Javaslang types. Future, Lazy, Option, Validation and Try have a `sequence()` method. We should take care that all sequence() methods return the same. That is not always possible (as described below) but the others should use Vector instead of List now.

* [x] Future.sequence() can stay as-is. It uses Stream in order to defer the blocking call of Future.get()
* [x] Option.sequence() uses Vector now instead of List
* [ ] Lazy.sequence() needs to use Vector (instead of List or Stream)
* [ ] Validation.sequence() needs to use Vector instead of List
* [ ] Try.sequence() needs to use Vector instead of List

Please use Vector for the sequence() methods of Lazy, Validation and Try.